pkg/apis/build/v1alpha1/build_source.go

```diff
@@ -0,0 +1,33 @@
+package v1alpha1
+
+// BuildSourceType enumerates build source type names.
+type BuildSourceType string
+
+// HTTP http build source name.
+const HTTP BuildSourceType = "http"
+
+// BuildSource remote artifact definition.
+type BuildSource struct {
+	// Name instance entry.
+	Name string `json:"name"`
+
+	// Type BuildSource type name.
+	Type BuildSourceType `json:"type"`
+
+	// URL remote artifact location.
+	URL string `json:"url"`
+
+	// TODO: define credentials to reach external artifacts.
+	// +optional
+	// Credentials *corev1.LocalObjectReference `json:"credentials"`
+
```

Comment:
In addition to per BuildSource creds, I could see the image used for the download varying on a per source basis

I'm good with maintaining the configurable global default you currently have, but an optional per BuildSource image override seems like a good idea.  It might not be as immediate with the one type (i.e. remote) but could be more prevalent as we add types.

Also, we do have this global default, per build override pattern in openshift build v1, and this config item feels similar to what we cover with that pattern.... for whatever that is worth.

I did go back to the EP @otaviof to see if this was discussed previously and I did not see anything.  Apologies if I missed it.  Or apologies if I missed making this suggestion back when reviewing the EP :-)

WDYT?

Reply:
I liked very much this suggestion, that's an elegant approach to extend the ability to support different types of remote artifacts, and also give users an opportunity to extend this behavior elegantly.

It was not part of the EP itself, so my suggestion is that we create a issue to discuss (and document) this approach, and make sure we tackle this change right after.

Reply:
Additionally, perhaps we could make this change part of #652 EP that @adambkaplan is working on.

Reply:
I'm fine with transferring the credentials granularity concern to https://github.com/shipwright-io/build/pull/652

In fact I already see there where it has been covered.  A yaml snippet from part of the proposed API there:

```
   - name: git
      type: Git
      git:
        url: https://github.com/shipwright-io/build.git
        ref: main
        credentials:
          name: git-creds
```

`credentials` is part of the specific source entry ... so finer granularity provided.

I consider my suggestion resolved @otaviof